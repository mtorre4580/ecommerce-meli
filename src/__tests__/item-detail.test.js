import React from 'react';
import ItemDetail from '../components/detail/item-detail/item-detail';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

describe('<ItemDetail />', () => {

    const categories = [
        'Electrónica, Audio y Video',
        'Audio',
        'Reproductores Portátiles',
        'iPod',
        'iPod touch',
        '32 GB'
    ];

    const item = {
        id:'MLA772322281',
        title:'Ipod Touch 4ta Generacion 32 Gb',
        price: {
            currency:'ARS',
            amount:680,
            decimals:null
        },
        picture:'http://mla-s2-p.mlstatic.com/738532-MLA25881108346_082017-I.jpg',
        condition:'used',
        location:'Capital Federal'
    };

    it('should render snapshot', () => {
        const component = renderer.create(<ItemDetail item={item} categories={categories} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should be equal categories, items props', () => {
        const wrapper = mount(<ItemDetail item={item} categories={categories} />);
        expect(wrapper.props().categories).toEqual(categories);
        expect(wrapper.props().item).toEqual(item);
    });
     
});



