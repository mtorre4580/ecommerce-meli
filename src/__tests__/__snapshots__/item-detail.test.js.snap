// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemDetail /> should render snapshot 1`] = `
<div
  className="container"
>
  <nav>
    <ol
      className="breadcrumb"
    >
      <li
        className="breadcrumb-item"
      >
        Electrónica, Audio y Video
      </li>
      <li
        className="breadcrumb-item"
      >
        Audio
      </li>
      <li
        className="breadcrumb-item"
      >
        Reproductores Portátiles
      </li>
      <li
        className="breadcrumb-item"
      >
        iPod
      </li>
      <li
        className="breadcrumb-item"
      >
        iPod touch
      </li>
      <li
        className="breadcrumb-item"
      >
        32 GB
      </li>
    </ol>
  </nav>
  <section
    className="item-detail"
  >
    <div
      className="item-detail-ctn"
    >
      <div
        className="item-detail-image"
      >
        <img
          alt="Ipod Touch 4ta Generacion 32 Gb"
          src="http://mla-s2-p.mlstatic.com/738532-MLA25881108346_082017-I.jpg"
        />
      </div>
      <div
        className="item-detail-header"
      >
        <p
          className="condition"
        >
          <span>
            Usado
             - 
          </span>
          <span>
            0
             vendidos
          </span>
        </p>
        <h2
          className="item-detail-title"
        >
          Ipod Touch 4ta Generacion 32 Gb
        </h2>
        <p
          className="price"
        >
          <span>
            $
          </span>
          680
        </p>
        <button
          className="btn btn-item-buy"
          type="button"
        >
          Comprar
        </button>
      </div>
    </div>
    <div
      className="item-description-ctn"
    >
      <h3>
        Descripción del producto
      </h3>
      <div
        className="item-description"
      />
    </div>
  </section>
</div>
`;
